src/nvim/mbyte.c

```diff
@@ -1992,6 +1992,31 @@ void utf_find_illegal(void)
   convert_setup(&vimconv, NULL, NULL);
 }
 
+/// @return  true if string "s" is a valid utf-8 string.
+/// When "end" is NULL stop at the first NUL.
+/// When "end" is positive stop there.
+bool utf_valid_string(const char_u *s, const char_u *end)
```

Comment:
we seriously don't have a function for this already? I would expect other parts of the code to be able to use this.

Reply:
The are other places that need to check for an invalid UTF-8 byte, but I guess they need more granular handling of invalid bytes, so this function isn't enough to satisfy the usage.